yaml_cpp_vendor-extras.cmake.in

```diff
@@ -16,3 +16,6 @@ set(yaml_cpp_vendor_LIBRARIES ${YAML_CPP_LIBRARIES})
 set(yaml_cpp_vendor_INCLUDE_DIRS ${YAML_CPP_INCLUDE_DIR})
 
 list(APPEND yaml_cpp_vendor_TARGETS yaml-cpp)
+if(WIN32)
+  set_target_properties(${YAML_CPP_LIBRARIES} PROPERTIES INTERFACE_COMPILE_DEFINITIONS YAML_CPP_DLL)
```

Comment:
I've identified the issue that was causing the build of `rosbag2_storage` to fail. It appears that `YAML_CPP_LIBRARIES` is now empty in this version of yaml-cpp. Using the target `yaml-cpp` directly fixes the issue.

```suggestion
  set_target_properties(yaml-cpp PROPERTIES INTERFACE_COMPILE_DEFINITIONS YAML_CPP_DLL)
```